gui/src/components/RemoteData.svelte

```diff
@@ -0,0 +1,36 @@
+<script lang="ts">
+  import type { RequestLifecycle } from '../lib/api';
+
+  type Item = $$Generic;
+  interface $$Slots {
+    pending: {};
+    success: { data: Item };
+    refetching: { data: Item };
+    error: { error: string };
+    default: {
+      data: Item;
+      error: string;
+    };
+  }
+  interface $$Props {
+    data: RequestLifecycle<Item>;
+  }
+
+  export let data: RequestLifecycle<Item>;
+</script>
+
+{#if data.stage === 'pending'}
+  <slot name="pending">Loading...</slot>
+{:else if data.stage === 'success'}
+  <slot name="success" data={data.data}>Missing success slot!</slot>
+{:else if data.stage === 'refetching'}
+  <slot name="refetching" data={data.data}>
+    <slot name="success" data={data.data}>
```

Comment:
if i'm reading this correctly, this is defaulting refreshing to success? intersting/neat

Reply:
Yes - I just discovered this, and it seemed like a good way to handle this sort of pattern of providing implementations at the right level of specificity so that the client code can decide to handle re-fetching differently than just having data.